services/datamanager/datasync/sync.go

```diff
@@ -81,23 +82,43 @@ func NewManager(identity string, client v1.DataSyncServiceClient, logger logging
 	cancelCtx, cancelFunc := context.WithCancel(context.Background())
 	logger.Debugf("Making new syncer with %d max threads", maxSyncThreads)
 	ret := syncer{
-		partID:             identity,
-		client:             client,
-		logger:             logger,
-		cancelCtx:          cancelCtx,
-		cancelFunc:         cancelFunc,
-		arbitraryFileTags:  []string{},
-		inProgress:         make(map[string]bool),
-		syncErrs:           make(chan error, 10),
-		syncRoutineTracker: make(chan struct{}, maxSyncThreads),
-		captureDir:         captureDir,
+		partID:            identity,
+		client:            client,
+		logger:            logger,
+		cancelCtx:         cancelCtx,
+		cancelFunc:        cancelFunc,
+		arbitraryFileTags: []string{},
+		inProgress:        make(map[string]bool),
+		syncErrs:          make(chan error, 10),
+
+		// syncRoutineTracker: make(chan struct{}, maxSyncThreads),
+		captureDir: captureDir,
 	}
 	ret.logRoutine.Add(1)
 	goutils.PanicCapturingGo(func() {
 		defer ret.logRoutine.Done()
 		ret.logSyncErrs()
 	})
 
+	for i := 0; i < maxSyncThreads; i++ {
```

Comment:
This is an great cleanup.

So we create 1000 goroutines and each reads from filesToSync, distributing the work efficiently across all of them. Perhaps we can set the default to 500 to account for rPis (or whatever test value you come up with)